haas/ext/switches/brocade.py

```diff
@@ -0,0 +1,179 @@
+        response = {}
+        for port in ports:
+            response[port] = filter(None, [self._get_native_vlan(port)]) \
+                             + self._get_vlans(port)
+        return response
+
+    def _get_mode(self, interface):
+        url = self._construct_url(interface, suffix='mode')
+        response = requests.get(url, auth=self._auth)
+        root = etree.fromstring(response.text)
+        mode = root.find(self._construct_tag('vlan-mode')).text
+        return mode
+
+    def _set_mode(self, interface, mode):
+        if mode in ['access', 'trunk']:
+            url = self._construct_url(interface, suffix='mode')
+            payload = '<mode><vlan-mode>%s</vlan-mode></mode>' % mode
+            requests.put(url, data=payload, auth=self._auth)
+        else:
+            raise AssertionError('Invalid mode')
+
+    def _get_vlans(self, interface):
+        try:
+            url = self._construct_url(interface, suffix='trunk')
+            response = requests.get(url, auth=self._auth)
+            root = etree.fromstring(response.text)
+            vlans = root.\
+                find(self._construct_tag('allowed')).\
+                find(self._construct_tag('vlan')).\
+                find(self._construct_tag('add')).text
+            return [('vlan/%s' % x, x) for x in vlans.split(',')]
+        except AttributeError:
+            return []
+
+    def _get_native_vlan(self, interface):
+        try:
+            url = self._construct_url(interface, suffix='trunk')
+            response = requests.get(url, auth=self._auth)
+            root = etree.fromstring(response.text)
+            vlan = root.find(self._construct_tag('native-vlan')).text
+            return ('vlan/native', vlan)
+        except AttributeError:
+            return None
+
+    def _add_vlan_to_trunk(self, interface, vlan):
+        self._set_mode(interface, 'trunk')
+        url = self._construct_url(interface, suffix='trunk/allowed/vlan')
+        payload = '<vlan><add>%s</vlan></vlan>' % vlan
+        requests.put(url, data=payload, auth=self._auth)
+
+    def _remove_vlan_from_trunk(self, interface, vlan):
+        url = self._construct_url(interface, suffix='trunk/allowed/vlan')
+        payload = '<vlan><remove>%s</remove></vlan>' % vlan
+        requests.put(url, data=payload, auth=self._auth)
+
+    def _set_native_vlan(self, interface, vlan):
+        self._set_mode(interface, 'trunk')
+        url = self._construct_url(interface, suffix='trunk')
+        payload = '<trunk><native-vlan>%s</native-vlan></trunk>' % vlan
+        requests.put(url, data=payload, auth=self._auth)
+
+    def _remove_native_vlan(self, interface):
+        url = self._construct_url(interface, suffix='trunk/native-vlan')
+        requests.delete(url, auth=self._auth)
+
+    def _construct_url(self, interface, suffix=None):
+        return '%(hostname)s/rest/config/running/interface/' \
+              '%(interface_type)s/%%22%(interface)s%%22/switchport/%(suffix)s' % {
```

Comment:
Would you put a comment in describing what the '%22' is about? I had to go look up what character that was. Using `urllib.quote` Might be a good alternative, but given that it's part of a larger string that could be awkward (though I think you can mark slashes as OK). I'll leave the urllib option to your descretion